sdk/messaging/azeventhubs/event_data_batch.go

```diff
@@ -166,3 +169,35 @@ func calcActualSizeForPayload(payload []byte) uint64 {
 
 	return uint64(vbin32Overhead + len(payload))
 }
+
+func newEventDataBatch(sender amqpwrap.AMQPSenderCloser, options *NewEventDataBatchOptions) (*EventDataBatch, error) {
+	if options == nil {
+		options = &NewEventDataBatchOptions{}
+	}
+
+	if options != nil && options.PartitionID != nil && options.PartitionKey != nil {
```

Comment:
The leading `options != nil` can be removed.

Reply:
Done.